--- a/src/alloc.c
+++ b/src/alloc.c
@@ -7197,20 +7197,6 @@
 		make_int (strings_consed));
 }
 
-#ifdef GNU_LINUX
-DEFUN ("malloc-info", Fmalloc_info, Smalloc_info, 0, 0, "",
-       doc: /* Report malloc information to stderr.
-This function outputs to stderr an XML-formatted
-description of the current state of the memory-allocation
-arenas.  */)
-  (void)
-{
-  if (malloc_info (0, stderr))
-    error ("malloc_info failed: %s", emacs_strerror (errno));
-  return Qnil;
-}
-#endif
-
 static bool
 symbol_uses_obj (Lisp_Object symbol, Lisp_Object obj)
 {
@@ -7555,9 +7541,6 @@
   defsubr (&Sgarbage_collect);
   defsubr (&Smemory_info);
   defsubr (&Smemory_use_counts);
-#ifdef GNU_LINUX
-  defsubr (&Smalloc_info);
-#endif
   defsubr (&Ssuspicious_object);
 
   Lisp_Object watcher;
